### DISCUSSION TITLE
how to use sweetalert

### DISCUSSION THREAD

**Discussion by bobwatcherx** [idom-team/idom#1010] [Question] [answered]
i read this web http://reactpy.dev/docs/guides/escape-hatches/javascript-components.html#dynamically-loaded-components .  but what about for other js libraries like sweetalert .  and i also want to use pdf js library later .  but my code sweetalert is not working

```python
from fastapi import FastAPI
from reactpy.backend.fastapi import configure
from reactpy import html,component,web
mui = web.module_from_template(
    # "react@^17.0.0",
    "react",
    # "@material-ui/core@4.12.4",
    # "sweetalert2",
    "@mui/material",
    fallback="waiting for my component .....",
)

ll = web.module_from_template(
    "react",

    "sweetalert2",
    fallback="waiting for my alert .....",

)
Alert = web.export(ll,"Swal")
Button = web.export(mui, "Button")


@component
def ViewButtonEvents():
    def showalert(event):
        Alert.fire({
          'title': 'Error!',
          'text': 'Do you want to continue',
          'icon': 'error',
          'confirmButtonText': 'Cool'
        })

    return html.div(
        Button(
            {
                "color": "primary",
                "variant": "contained",
                "onClick": showalert,
            },
            "Click Me!",
        ),
    )

app = FastAPI()
configure(app,ViewButtonEvents)
```

**Comment by bobwatcherx** [comment]
I have installed all packages with npm in my project.  for material ui it works but sweetalert doesn't

**Comment by jordan-sussman** [comment]
@bobwatcherx could it be because `showalert(event)` accepts an event but one isn't passed for the Button's `onClick: showalert`?

**Reply by rmorshea** [reply]
That's definitely possible. My guess is that there's something else going wrong here. `module_from_template`, unfortunately is pretty finicky, so my guess is that's the issue.

**Reply by Archmonger** [reply, marked as answer]
`module_from_template` is scheduled for removal due to random technical bugs. Your issue is likely related.

See https://github.com/reactive-python/reactpy/issues/965 for more details

**Comment by rmorshea** [comment]
Are there any logs in your browser's dev console or your terminal?
